src/cockpit/channel.py

```diff
@@ -168,6 +168,7 @@ async def write(self, data):
             self.out_sequence += len(data)
 
             while self.out_window < self.out_sequence:
+                self.send_control(command='ping', sequence=self.out_sequence)
```

Comment:
I think this isn't right.  Waiting until we're already going to block is going to result in unnecessary latency.

We rather want to send a ping, unconditionally, for every `n` bytes we send.

Reply:
the C bridge has these constants:

```C
/* Every 16K Send a ping */
#define  CHANNEL_FLOW_PING        (16L * 1024L)

/* Allow up to 1MB of data to be sent without ack */
#define  CHANNEL_FLOW_WINDOW       (2L * 1024L * 1024L)
```

and implements the following algorithm for deciding when to send:

```C
      /* Every CHANNEL_FLOW_PING bytes we send a ping; also when applying back
       * pressure as there is otherwise nothing more to send and generate pings for */
      if ((out_sequence / CHANNEL_FLOW_PING != priv->out_sequence / CHANNEL_FLOW_PING) || trigger_pressure)
        {
```

Reply:
Alright!

Reply:
I don't think we need something like `trigger_pressure`.